### PR TITLE
[snapshot] add headless option

### PR DESCRIPTION
**Pull request by ChaosCoder**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

When snapshotting simulators we are currently not showing the simulator windows. This has UI side effects, one of the more common problems is that a simulated user location (via Simulators `Features -> Location -> *`) will not be shown. This is especially cumbersome for apps that display maps, as the "blue dot" never appears. #16440 #10381 #12682 #1678

This PR introduces a new `headless` option. If `headless` is disabled, snapshot will take care of opening the simulator after it has been killed by snapshot and applying some simulator settings.

### Description

Introducing a new `headless` parameter and documenting it. The `headless` parameter is `true` by default, leaving the old behavior as default. After killing the simulator and applying the settings (like localization, appearance, slide to type), the simulator will now be opened again, if `headless` is `false`.

### Testing Steps

Specify the `headless` parameter when invoking snapshot and set it to `false`, or leave it off completely to get the old (default) behavior. Set it to `true` and the Simulator.app will be launched before the snapshot tests are invoked.

### CODE REVIEW

File: snapshot/lib/snapshot/simulator_launchers/simulator_launcher_base.rb [hunk 1]
```diff
@@ -76,6 +76,11 @@ def prepare_simulators_for_launch(device_types, language: nil, locale: nil)
           disable_slide_to_type(type)
         end
       end
+
+      unless launcher_config.headless
+        simulator_path = File.join(Helper.xcode_path, 'Applications', 'Simulator.app')
+        Helper.backticks("open -a #{simulator_path} -g", print: FastlaneCore::Globals.verbose?)
+      end
```
Comment (janpio) [inline snapshot/lib/snapshot/simulator_launchers/simulator_launcher_base.rb:83]: That easy? Nice.

Comment (ChaosCoder) [reply snapshot/lib/snapshot/simulator_launchers/simulator_launcher_base.rb:83]: Yeah, that easy. As I said, snapshot kills the Simulator.app process and never starts it again. The workaround was to manually open the Simulator.app after snapshot closed it, so now we just do that automatically. 😅

**Review by janpio — APPROVED** [review]

**Review by joshdholtz — COMMENTED** [review]
@ChaosCoder Hey 👋 Code looks great! Just a question about behavior here 😊

I pulled this down and ran it with both `headless: true` and `headless: false`. The simulators opened for me both times. Is that not what I should be expecting? Or is headless a different meaning here? 🤔

**Comment by ChaosCoder** [comment]
> @ChaosCoder Hey 👋 Code looks great! Just a question about behavior here 😊
> 
> I pulled this down and ran it with both `headless: true` and `headless: false`. The simulators opened for me both times. Is that not what I should be expecting? Or is headless a different meaning here? 🤔

Specifying `headless: false` should not open the simulators. Can you run it with verbose enabled? Was the Simulator.app process killed by snapshot (e.g. is the Simulators.app running)? It might be, that it was never killed because snapshot did not apply any special settings to the simulators (just a guess).

**Comment by ChaosCoder** [comment]
@googlebot I consent.

**Comment by joshdholtz** [comment]
@googlebot I consent.

**Comment by joshdholtz** [comment]
> > @ChaosCoder Hey 👋 Code looks great! Just a question about behavior here 😊
> > I pulled this down and ran it with both `headless: true` and `headless: false`. The simulators opened for me both times. Is that not what I should be expecting? Or is headless a different meaning here? 🤔
> 
> Specifying `headless: false` should not open the simulators. Can you run it with verbose enabled? Was the Simulator.app process killed by snapshot (e.g. is the Simulators.app running)? It might be, that it was never killed because snapshot did not apply any special settings to the simulators (just a guess).

Yeah yeah yeah, this is all good now! I had to attempt to force quit simulators a few times 🤷‍♂️

**Review by joshdholtz — APPROVED** [review]
Got this working! Thanks for adding this and waiting for me to figure out how to test it properly ❤️ Really appreciate the contribution!

**Comment by hactar** [comment]
@ChaosCoder @joshdholtz Could we have this for scan too? We have the same issue when running our tests - User Location simulation doesn't work reliably in headless mode...

**Pull request merged**